backend/src/users/users.service.ts

```diff
@@ -18,5 +19,19 @@ export class UsersService {
         return await this.usersRepository.findOne({ username: username });
     }
 
-    // Create, update, delete users: Do it through DB directly?
+    // Create must be done through service because of hashing functionality
+    // Is there any better way? Feel free to suggest
+    async create(usersEntity: UsersEntity): Promise<any> {
+        // Technically, usersData.hashedPassword here isn't hashed yet when it's sent through the form
+        var hash = AuthService.hashPasswordSync(usersEntity.hashedPassword);
```

Comment:
I suggest hashing the password in controller level before sending `usersEntity` to the service but don't worry, this is ok just my suggestion no need to change.